core/shared/src/main/scala/zio/package.scala

```diff
@@ -42,6 +42,16 @@ package object zio extends EitherCompat with PlatformSpecific with VersionSpecif
   type Ref[A]      = ZRef[Nothing, Nothing, A, A]
   type ERef[+E, A] = ZRef[E, E, A, A]
 
+  type ZLens[-S, +T, +A, -B]      = (S => A, B => S => T)
```

Comment:
We could be more polymorphic here, allowing each of these optics to have a get error and a set error here, but then we would need some constructors that make it easy for users to lift simple optics into these more polymorphic types.

Reply:
This could also allow us to support some other types of optics such as partial isomorphisms from ZIO Config.